### PR TITLE
qemu_monitor.py: Add bitmap and live backup command.

### PR DESCRIPTION
**Pull request by vivianQizhu**
Add bitmap and live backup command.

Signed-off-by: Qianqian Zhu <qizhu@redhat.com>

### CODE REVIEW

**Comment by vivianQizhu** [comment]
03:48:18 INFO | Create bitmap for drive_image1
03:48:18 DEBUG| (monitor qmp1) Sending command 'block-dirty-bitmap-add'
03:48:18 DEBUG| Send command: {'execute': 'block-dirty-bitmap-add', 'arguments': {'node': u'drive_image1', 'name': 'bitmap0', 'granularity': 131072}, 'id': 'kASgah7F'}
03:48:18 INFO | Create full backup for drive_image1
03:48:18 DEBUG| (monitor qmp1) Sending command 'drive-backup'
03:48:18 DEBUG| Send command: {'execute': 'drive-backup', 'arguments': {'device': u'drive_image1', 'mode': 'absolute-paths', 'sync': 'full', 'target': '/usr/share/avocado/data/avocado-vt/images/fullbackup.qcow2', 'format': 'qcow2'}, 'id': '2NJyboHx'}
03:48:18 INFO | [qemu output] Formatting '/usr/share/avocado/data/avocado-vt/images/fullbackup.qcow2', fmt=qcow2 size=21474836480 encryption=off cluster_size=65536 lazy_refcounts=off refcount_bits=16

File: virttest/qemu_monitor.py [hunk 1]
```diff
@@ -198,7 +198,7 @@ def __init__(self, vm, name, filename, suppress_exceptions=False):
         self._passfd = None
         self._supported_cmds = []
         self.debug_log = False
-        self.log_file = "%s-%s.log" % (name, vm.name)
+        self.log_file = os.path.basename(self.filename + ".log")
```
Comment (xutian) [inline virttest/qemu_monitor.py]: why changed it ?

File: virttest/qemu_monitor.py [hunk 2]
```diff
@@ -1258,29 +1258,6 @@ def get_migrate_cache_size(self):
...
         return value
 
-    def set_migrate_parameter(self, parameter, value):
-        """
-        Set parameters of migrate.
-
-        :param parameter: the parameter which need to set
-        :param value: the value of parameter
-        """
-        cmd = "migrate_set_parameter"
```
Comment (xutian) [inline virttest/qemu_monitor.py]: why ?

File: virttest/qemu_monitor.py [hunk 3]
```diff
@@ -1258,29 +1258,6 @@ def get_migrate_cache_size(self):
...
-        self.verify_supported_cmd(cmd)
-        cmd += " %s %s" % (parameter, value)
-        return self.cmd(cmd)
-
-    def get_migrate_parameter(self, parameter):
-        """
-        Get the parameter value. e.g. cpu-throttle-initial: 30
-
-        :param parameter: the parameter which need to get
```
Comment (xutian) [inline virttest/qemu_monitor.py]: why ?

File: virttest/qemu_monitor.py [hunk 4]
```diff
@@ -2322,30 +2299,6 @@ def get_migrate_cache_size(self):
...
         return self.query("migrate-cache-size")
 
-    def set_migrate_parameter(self, parameter, value):
-        """
-        Set the parameters of migrate.
-
-        :param parameter: the parameter which need to set.
-        :param value: the value of parameter
-        """
-        cmd = "migrate-set-parameters"
```
Comment (xutian) [inline virttest/qemu_monitor.py]: same question

**Comment by vivianQizhu** [comment]
@xutian  Thanks, close it and open a new one: 
https://github.com/avocado-framework/avocado-vt/pull/800
Please help review, thanks.

**Comment by xutian** [comment]
@vivianQizhu no need to create a new PR, just need to  pull fork a new branch base on your master then cherry-pick your commit to new branch, then drop live_backup branch and rename new branch to live_backup and push live_backup branch with --force option is ok

**Comment by vivianQizhu** [comment]
@xutian Thanks a lot, but now hard for me to do so, since live_backup branch has been already overwrite, this commit is lost, I will follow your steps next time if I meet this situation again.

**Pull request closed without merging**